config/environments/development.rb

```diff
@@ -78,18 +77,26 @@
       next if addrinfo.ip_address == "127.0.0.1"
 
       ip = IPAddr.new(addrinfo.ip_address).mask(24)
-
       Logger.new($stdout).info "Adding #{ip.inspect} to config.web_console.permissions"
 
       config.web_console.permissions = ip
     end
   end
 
+  # Annotate rendered view with file names.
+  config.action_view.annotate_rendered_view_with_filenames = true
```

Comment:
This adds html comments to the markup that let you know that this segment of html is partial A and this other segment is from the layout.